docs/api.adoc

```diff
@@ -22,6 +22,148 @@ Enonic XP comes with a large number of build in JavaScript libraries for develop
 Some are straight helper classes, other provide interaction with the data in the site.
 Please check out all of them in the submenu.
 
+== TypeScript definitions
+
+=== Install
+
+To add types for corresponding library use NPM (or similar package manager) and install them as any other depencency. All types can be found under https://www.npmjs.com/org/enonic-types[@enonic-types] organization on NPM.
+
+==== Libraries
+
+For example, you have a `lib-auth` gradle dependecy:
+
+```gradle
+dependencies {
+  include "com.enonic.xp:lib-auth:${xpVersion}"
+}
+```
+
+To add types, just use the same library's name:
+
+```bash
+npm i --save-dev @enonic-types/lib-auth
+```
+
+==== Global
+
+Along with library types it may be useful to install definitions for XP global objects (`app`, `__`, `log`) and functions (`resolve`, `require`):
```

Comment:
```suggestion
Along with library types it may be useful to install definitions for global objects (`app`, `__`, `log`) and functions (`resolve`, `require`):
```